core/src/main/scala/org/apache/spark/sql/rapids/tool/AppBase.scala

```diff
@@ -163,6 +163,30 @@ abstract class AppBase(
     }
   }
 
+  /**
+   * Calculates total core ms which is the sum over executor core ms. Core ms per executor
+   * is computed as executor duration (ms) multiplied by num of cores in the executor.
+   */
+  def calculateTotalCoreMs(): Long = {
+    var totalCoreMs: Long = 0L
+    executorIdToInfo.foreach { case(_, eInfo) =>
+      val eStartTime = eInfo.addTime
+      var eEndTime = eInfo.removeTime
+      if (eEndTime == 0L) {
+        getAppEndTime match {
+          case Some(appEndTime) =>
+            eEndTime = appEndTime
+          case None =>
+            logInfo("Unable to find either executor or app end time: " +
+              "setting executor duration to 0")
+            eEndTime = eStartTime
+        }
+      }
+      totalCoreMs += (eEndTime - eStartTime) * eInfo.totalCores
```

Comment:
any reason not to convert to seconds before doing the multiplication here?

Reply:
updated this calculation, PTAL